### PR TITLE
add R 4.1.0 with foss/2021a to EESSI pilot 2021.12

### PR DESCRIPTION
**Pull request by smoretti**
Add `R-4.1.0-foss-2021a.eb` in the EESSI **2021.12** pilot

edit: checklist:
* [x] `aarch64/generic` (done, [ingested](https://github.com/EESSI/staging/pull/123))
* [x] `aarch64/graviton2` (done, [ingested](https://github.com/EESSI/staging/pull/116))
* [x] `aarch64/graviton3` (done, [ingested](https://github.com/EESSI/staging/pull/117))
* [x] `ppc64le/generic` (done, [ingested](https://github.com/EESSI/staging/pull/125))
* [x] `ppc64le/power9le` (done, [ingested](https://github.com/EESSI/staging/pull/124))
* [x] `x86_64/generic` (done, [ingested](https://github.com/EESSI/staging/pull/120))
* [x] `x86_64/amd/zen2` (done, [ingested](https://github.com/EESSI/staging/pull/118))
* [x] `x86_64/amd/zen3` (done, [ingested](https://github.com/EESSI/staging/pull/119))
* [x] `x86_64/intel/haswell` (done, [ingested](https://github.com/EESSI/staging/pull/121))
* [x] `x86_64/intel/skylake_avx512` (done,[ingested](https://github.com/EESSI/staging/pull/122))

### CODE REVIEW

File: EESSI-pilot-install-software.sh [hunk 1]
```diff
@@ -356,6 +356,12 @@ fail_msg="Installation of WRF failed, that's unexpected..."
 OMPI_MCA_pml=ucx UCX_TLS=tcp $EB WRF-3.9.1.1-foss-2020a-dmpar.eb -r --include-easyblocks-from-pr 2648
 check_exit_code $? "${ok_msg}" "${fail_msg}"
 
+echo ">> Installing R 4.1.0 (better be patient)..."
+ok_msg="R installed, wow!"
+fail_msg="Installation of R failed, so sad..."
+$EB --from-pr 16386 R-4.1.0-foss-2021a.eb --robot --parallel-extensions-install --experimental
```
Comment (boegel) [inline EESSI-pilot-install-software.sh:362]: @smoretti Instructing EasyBuild to download the `R` easyconfig from https://github.com/easybuilders/easybuild-easyconfigs/pull/16386 is problematic currently, it fails with:

```
ERROR: Couldn't find path to patched file /tmp/eb-90906wdf/files_pr16386/h/hyperopt/hyperopt-0.2.4-intel-2019b-Python-3.7.4-Java-1.8.eb
```

That's a result of the large number of easyconfigs being touched in this PR, and the changes that were made in https://github.com/easybuilders/easybuild-easyconfigs/pull/16697 (so `hyperopt-0.2.4-intel-2019b-Python-3.7.4-Java-1.8.eb` no longer exists in the `develop` branch).

To some extent, that's a bug in EasyBuild (maybe), but we'll need to work around it.

The most recent change to `R-4.1.0-foss-2021a.eb` that really matters was made in https://github.com/easybuilders/easybuild-easyconfigs/pull/16011 (checksum fix), so let's use that instead?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
With the current installations in EESSI pilot 2021.12:

57 out of 95 required modules missing

<details>

```
* pixman/0.40.0-GCCcore-10.3.0 (pixman-0.40.0-GCCcore-10.3.0.eb)
* Bison/3.7.6-GCCcore-10.3.0 (Bison-3.7.6-GCCcore-10.3.0.eb)
* PCRE2/10.36-GCCcore-10.3.0 (PCRE2-10.36-GCCcore-10.3.0.eb)
* libunwind/1.4.0-GCCcore-10.3.0 (libunwind-1.4.0-GCCcore-10.3.0.eb)
* libpng/1.6.37-GCCcore-10.3.0 (libpng-1.6.37-GCCcore-10.3.0.eb)
* nettle/3.7.2-GCCcore-10.3.0 (nettle-3.7.2-GCCcore-10.3.0.eb)
* NASM/2.15.05-GCCcore-10.3.0 (NASM-2.15.05-GCCcore-10.3.0.eb)
* libjpeg-turbo/2.0.6-GCCcore-10.3.0 (libjpeg-turbo-2.0.6-GCCcore-10.3.0.eb)
* Brotli/1.0.9-GCCcore-10.3.0 (Brotli-1.0.9-GCCcore-10.3.0.eb)
* freetype/2.10.4-GCCcore-10.3.0 (freetype-2.10.4-GCCcore-10.3.0.eb)
* Ninja/1.10.2-GCCcore-10.3.0 (Ninja-1.10.2-GCCcore-10.3.0.eb)
* fontconfig/2.13.93-GCCcore-10.3.0 (fontconfig-2.13.93-GCCcore-10.3.0.eb)
* Mako/1.1.4-GCCcore-10.3.0 (Mako-1.1.4-GCCcore-10.3.0.eb)
* Meson/0.58.0-GCCcore-10.3.0 (Meson-0.58.0-GCCcore-10.3.0.eb)
* GLib/2.68.2-GCCcore-10.3.0 (GLib-2.68.2-GCCcore-10.3.0.eb)
* jbigkit/2.1-GCCcore-10.3.0 (jbigkit-2.1-GCCcore-10.3.0.eb)
* gzip/1.10-GCCcore-10.3.0 (gzip-1.10-GCCcore-10.3.0.eb)
* X11/20210518-GCCcore-10.3.0 (X11-20210518-GCCcore-10.3.0.eb)
* libdrm/2.4.106-GCCcore-10.3.0 (libdrm-2.4.106-GCCcore-10.3.0.eb)
* cairo/1.16.0-GCCcore-10.3.0 (cairo-1.16.0-GCCcore-10.3.0.eb)
* libglvnd/1.3.3-GCCcore-10.3.0 (libglvnd-1.3.3-GCCcore-10.3.0.eb)
* Tk/8.6.11-GCCcore-10.3.0 (Tk-8.6.11-GCCcore-10.3.0.eb)
* LLVM/11.1.0-GCCcore-10.3.0 (LLVM-11.1.0-GCCcore-10.3.0.eb)
* lz4/1.9.3-GCCcore-10.3.0 (lz4-1.9.3-GCCcore-10.3.0.eb)
* zstd/1.4.9-GCCcore-10.3.0 (zstd-1.4.9-GCCcore-10.3.0.eb)
* Mesa/21.1.1-GCCcore-10.3.0 (Mesa-21.1.1-GCCcore-10.3.0.eb)
* libGLU/9.0.1-GCCcore-10.3.0 (libGLU-9.0.1-GCCcore-10.3.0.eb)
* LibTIFF/4.2.0-GCCcore-10.3.0 (LibTIFF-4.2.0-GCCcore-10.3.0.eb)
* Xvfb/1.20.11-GCCcore-10.3.0 (Xvfb-1.20.11-GCCcore-10.3.0.eb)
* NLopt/2.7.0-GCCcore-10.3.0 (NLopt-2.7.0-GCCcore-10.3.0.eb)
* ICU/69.1-GCCcore-10.3.0 (ICU-69.1-GCCcore-10.3.0.eb)
* libogg/1.3.4-GCCcore-10.3.0 (libogg-1.3.4-GCCcore-10.3.0.eb)
* FLAC/1.3.3-GCCcore-10.3.0 (FLAC-1.3.3-GCCcore-10.3.0.eb)
* libvorbis/1.3.7-GCCcore-10.3.0 (libvorbis-1.3.7-GCCcore-10.3.0.eb)
* libsndfile/1.0.31-GCCcore-10.3.0 (libsndfile-1.0.31-GCCcore-10.3.0.eb)
* UDUNITS/2.2.28-GCCcore-10.3.0 (UDUNITS-2.2.28-GCCcore-10.3.0.eb)
* Szip/2.1.1-GCCcore-10.3.0 (Szip-2.1.1-GCCcore-10.3.0.eb)
* GSL/2.7-GCC-10.3.0 (GSL-2.7-GCC-10.3.0.eb)
* GLPK/5.0-GCCcore-10.3.0 (GLPK-5.0-GCCcore-10.3.0.eb)
* Ghostscript/9.54.0-GCCcore-10.3.0 (Ghostscript-9.54.0-GCCcore-10.3.0.eb)
* nodejs/14.17.0-GCCcore-10.3.0 (nodejs-14.17.0-GCCcore-10.3.0.eb)
* JasPer/2.0.28-GCCcore-10.3.0 (JasPer-2.0.28-GCCcore-10.3.0.eb)
* HDF5/1.10.7-gompi-2021a (HDF5-1.10.7-gompi-2021a.eb)
* LittleCMS/2.12-GCCcore-10.3.0 (LittleCMS-2.12-GCCcore-10.3.0.eb)
* ImageMagick/7.0.11-14-GCCcore-10.3.0 (ImageMagick-7.0.11-14-GCCcore-10.3.0.eb)
* GEOS/3.9.1-GCC-10.3.0 (GEOS-3.9.1-GCC-10.3.0.eb)
* PCRE/8.44-GCCcore-10.3.0 (PCRE-8.44-GCCcore-10.3.0.eb)
* libgit2/1.1.0-GCCcore-10.3.0 (libgit2-1.1.0-GCCcore-10.3.0.eb)
* PROJ/8.0.1-GCCcore-10.3.0 (PROJ-8.0.1-GCCcore-10.3.0.eb)
* libiconv/1.16-GCCcore-10.3.0 (libiconv-1.16-GCCcore-10.3.0.eb)
* Doxygen/1.9.1-GCCcore-10.3.0 (Doxygen-1.9.1-GCCcore-10.3.0.eb)
* netCDF/4.8.0-gompi-2021a (netCDF-4.8.0-gompi-2021a.eb)
* libgeotiff/1.6.0-GCCcore-10.3.0 (libgeotiff-1.6.0-GCCcore-10.3.0.eb)
* libtirpc/1.3.2-GCCcore-10.3.0 (libtirpc-1.3.2-GCCcore-10.3.0.eb)
* HDF/4.2.15-GCCcore-10.3.0 (HDF-4.2.15-GCCcore-10.3.0.eb)
* GDAL/3.3.0-foss-2021a (GDAL-3.3.0-foss-2021a.eb)
* R/4.1.0-foss-2021a (R-4.1.0-foss-2021a.eb)
```
</details>

File: EESSI-pilot-install-software.sh (hunk 1 above)
Comment (smoretti) [reply EESSI-pilot-install-software.sh]: Done

Comment (smoretti) [reply EESSI-pilot-install-software.sh]: I thought using a PR number **and** an eb filename should avoid using all the other eb files from the PR

**Review by boegel — CHANGES_REQUESTED** [review]
@smoretti We need to install `X11` via a PR too, or the installation fails due to a missing patch.

Fixed in https://github.com/smoretti/EESSI_software-layer/pull/1

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
(finally) built and ingested for all CPU targets, so merging

**Pull request merged**